### ISSUE TITLE
Indented code lines miss line breaks

### ISSUE THREAD

**Issue by vaygr** [ncornette/Python-Markdown-Editor#9]
I'm trying to make the following markdown code work properly:

`````` markdown
**NOTES:**

1. one
2. two

    ```
    code line 1
    ```

    text

    ```
    code line 2
    code line 3
    ```

3. three
``````

For some strange reason `code line 2` and `code line 3` are joined without any line break like this:

```
code line 2 code line 3
```

instead of showing 2 separate lines. On github and [dillinger](http://dillinger.io/) it works properly though.

Setup is the following:
- python 2.7.9
- python-markdown 2.6.6
- pygments 2.0.2
- bottle 0.12.9

**Comment by ncornette** [comment]
Hi, I can confirm apparently python-markdown doesn't handle indented code blocks, 

Thank you for reporting the issue.

**Comment by vaygr** [comment]
Interesting, maybe there's a chance to add support for [python-markdown2](https://pypi.python.org/pypi/markdown2)?

**Comment by vaygr** [comment]
Tested with python-markdown2, same issue.

**Comment by ncornette** [comment]
An option would be to add an extension at markdown initialization the same way I did for strikethrough. 

A project exists which contains almost all GFM extensions to python markdown including strikethrough : https://github.com/Zopieux/py-gfm 

Unfortunately it doesn't seem to support indented codeblocks...

At least this project could improve support for GFM, if it can be confirmed it doesn't support indented codeblocks, making a fork could also be an option.

**Comment by vaygr** [comment]
Wait a sec, isn't [this](https://pythonhosted.org/Markdown/extensions/nl2br.html) what we're looking for?

**Comment by vaygr** [comment]
Also [this project](https://github.com/facelessuser/pymdown-extensions) might be helpful.

**Comment by vaygr** [comment]
[This project](https://github.com/facelessuser/pymdown-extensions) might be helpful. I've managed to make it working using [superfences extension](http://facelessuser.github.io/pymdown-extensions/extensions/superfences/) with modification from [this PR](https://github.com/ncornette/Python-Markdown-Editor/pull/11) I've just created by executing this:

```
markdown_edit -x pymdownx.superfences test.md
```

Should `pymdownx.superfences` be included *by default* introducing yet another dependency or should the above-mentioned PR be merged in order to make overriding work correctly?

**Comment by ncornette** [comment]
If `pymdownx.superfences` extension makes rendering closer to GFM, I would recommend you include it directly into defaults extensions by updating `MARKDOWN_EXT` constant

**Comment by vaygr** [comment]
@ncornette it should be pretty close. The closest one would probably be [pymdownx.github](http://facelessuser.github.io/pymdown-extensions/extensions/github/). But the only concern I have is yet another dependency just for ability to use GFM-like code blocks, while with optional `-x` argument there's more freedom for an end user. But it's not critical, I agree.

By the way, unfortunately I couldn't make the [tasklist extension](http://facelessuser.github.io/pymdown-extensions/extensions/tasklist/) (which is included in `pymdownx.github`) work properly -- it concatenates list and check chars. Do you have any idea on what would cause that?

**Comment by vaygr** [comment]
Oh, I've figured out it was just missing `task-list` class. With this patch it works as expected:

```diff
--- markdown_editor/css/markdown.css.orig	2016-06-06 18:24:21.000000000 -0400
+++ markdown_editor/css/markdown.css	2017-01-17 20:49:16.872820211 -0500
@@ -113,7 +113,7 @@
 padding-left:30px
 }
 
-.markdown-body ul.no-list,.markdown-body ol.no-list{
+.markdown-body ul.no-list,.markdown-body ul.task-list,.markdown-body ol.no-list,.markdown-body ol.task-list{
 list-style-type:none;
 padding:0
 }
```

So what do you think? Along with this patch and additional entry of `pymdownx.github` in `MARKDOWN_EXT` should we introduce yet another dependency for better GFM support?

**Comment by ncornette** [comment]
Hey Vaygr !

It seems you found solutions to improve GFM rendering, don't hesitate to update your PR with your improvements, I will check the result before merging!

Let me know if you agree to proceed this way, I will eventually update the version and publish a new release on PIP.

Thank you for your work !

**Comment by vaygr** [comment]
@ncornette absolutely!

I've pushed respective commits to the above-mentioned PR for your review.

**Comment by vaygr** [comment]
Verified in recently released **1.0.3**. Works like a charm. Thanks!

p.s. therefore closing this.
